src/record_syscall.cc

```diff
@@ -3976,6 +3979,7 @@ static Switchable rec_prepare_syscall_arch(RecordTask* t,
       return PREVENT_SWITCH;
 
     case Arch::arch_prctl:
+      DEBUG_ASSERT(t->arch() == x86 || t->arch() == x86_64);
```

Comment:
Likewise this shouldn't be necessary because the arch_prctl syscallno is -121 on ARM.

Reply:
I thought it would be useful to be defensive here, just in case it gets added on another architecture. But I guess such a commit would also add a test that would catch any issue, so I'll get rid of this.